### PR TITLE
add oo design practise training

### PR DESCRIPTION
**Pull request by jukecraft**

### CODE REVIEW

File: _data/trainings.yml [hunk 1]
```diff
@@ -50,7 +50,13 @@
...
-- id: akka 
+- id: akka
   name: Scala Deep Dive with Akka
   description: For developers that want to deepen their Scala knowledge using its most popular toolkit, Akka
   page: /services/training/scala-deep-dive-with-akka
+
+- id: oo_design
+  name: Object Oriented Design Practise
+  author: Franziska Sauerwein
+  description: For experienced developers that want to practise creating object oriented designs.
```
Comment (dhatanian) [inline _data/trainings.yml]: _who_ want to practise

Comment (dhatanian) [reply _data/trainings.yml]: Maybe put a bit of an improvement notion here? "experienced developers who want to improve their design skills in object-oriented languages" ?

See Sandro's crafted design course description

File: services/training/oo-design.html [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+layout: training
+training: Object Oriented Design Practise
+---
+
+<!-- Course summary -->
+<div class="row text-center margin-bottom-60">
+    <div class="col-md-6 col-md-offset-3">
+        <h1>Object Oriented Design Practise</h1>
+        <h4 class="margin-bottom-20">Created by Franziska Sauerwein</h4>
+        <p class="margin-bottom-40 training-description"><strong>For experienced developers that want to practise creating object oriented designs.</strong> Design objects and their relationships using the principles of OO.
```
Comment (dhatanian) [inline services/training/oo-design.html]: Same: "who want"

Comment (dhatanian) [reply services/training/oo-design.html]: Also same comment on the improvement part

File: services/training/oo-design.html [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        <div class="row margin-bottom-20">
+            <h3>Course Duration</h3>
+	    <p>2 days</p>
+        </div>
+
+	<div class="row margin-bottom-20">
+            <h3>Why attend?</h3>
+            <ul>
+                <li>Practise the use of the most important OO design tools</li>
+                <li>Use Test Driven Development, Class-Relationship-Collaborator (CRC) cards and sequence diagrams to design</li>
```
Comment (dhatanian) [inline services/training/oo-design.html]: .. and sequence diagrams to design --> to design what?

File: services/training/oo-design.html [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                <li>Practise separating responsibilities into objects</li>
```
Comment (dhatanian) [inline services/training/oo-design.html]: between instead of into?

File: services/training/oo-design.html [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                <li>Fill your toolbelt with different approaches to software design to be able to pick the right one according to the context</li>
+                <li>Understand and communicate benefits and drawbacks of different desing decisions</li>
+                <li>Keep the design flexible and iterate, verifying your assumptions with code</li>
+            </ul>
+        </div>
+
+        <div class="row margin-bottom-20">
+            <h3>Content</h3>
+            <ul>
+                <li>Recap of design principles</li>
```
Comment (dhatanian) [inline services/training/oo-design.html]: Isn't recap a shortened word? Maybe put the full word

File: services/training/oo-design.html [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                <li>Use CRC cards as a design tool</li>
+                <li>Applying SOLID principles to object oriented design</li>
```
Comment (dhatanian) [inline services/training/oo-design.html:46]: Pick either the -ing noun form or the infinitive, but stay consistent

File: services/training/oo-design.html [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                <li>Using sequence and class relationship diagrams to design</li>
+                <li>Using tests to drive the design</li>
+                <li>Structure problems for iterative solutions</li>
+                <li>Verify your design assumptions with code</li>
+            </ul>
+        </div>
+
+        <div class="row margin-bottom-20">
+            <h3>Audience</h3>
+            <p>Software developers that:</p>
```
Comment (dhatanian) [inline services/training/oo-design.html]: Software developers who: ?

File: services/training/oo-design.html [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            <ul>
+                <li>Have experience in developing software commercially</li>
+                <li>Are familiar with at least one Object Oriented Language (such as Java and C#)</li>
```
Comment (dhatanian) [inline services/training/oo-design.html]: Java or C#

File: services/training/oo-design.html [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                <li>Have basic experience in Object Oriented Design</li>
+                <li>Are comfortable using TDD to build software</li>
+                <li>Understand design principles of OO such as SOLID and the 4 rules of simple design</li>
```
Comment (dhatanian) [inline services/training/oo-design.html]: Add links here to both concepts. People might know the concept but not the names. No need for them to think "ah no I don't know this, I won't take the course"

File: services/training/oo-design.html (hunk 7 above)
Comment (dhatanian) [inline services/training/oo-design.html]: developers and architects?

File: services/training/oo-design.html [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+                    <ul>
+                        <li>an object oriented language</li>
+                        <li>Their favourite IDE</li>
+                        <li>A unit testing framework</li>
+                    </ul>
+                </li>
+                <li>Be able to create projects, build source code and run test cases in their development environment
+                </li>
+            </ul>
+            <p>In addition a distributed version control system such as Git is desirable.</p>
```
Comment (dhatanian) [inline services/training/oo-design.html]: Is "distributed" important? SVN would serve us right here I guess?

**Review by dhatanian — COMMENTED** [review]
Hey Franzi, good course description, just a few suggestions added

File: _data/trainings.yml (hunk 1 above)
Comment (jukecraft) [reply _data/trainings.yml]: that's who I copied from 😁

Comment (dhatanian) [reply _data/trainings.yml]: Yeah I had that impression. If you look at his message, he offers people to improve, you "only" offer to practise.

Comment (dhatanian) [reply _data/trainings.yml]: Forget that comment, I missed your 👍  on my original comment

File: services/training/oo-design.html (hunk 7 above)
Comment (jukecraft) [reply services/training/oo-design.html]: I seem to be consistently wrong.
I checked, I copied it from Sandro 😄

File: services/training/oo-design.html (hunk 10 above)
Comment (jukecraft) [reply services/training/oo-design.html]: I think I'll remove it, I don't see a reason to put something in version control.

**Pull request merged**